File: backend/src/main/java/com/course/platform/payment/dto/PaymentDetailDto.java
package com.course.platform.payment.dto;

import java.math.BigDecimal;
import java.time.OffsetDateTime;

import com.course.platform.course.dto.CourseSummaryDto;
import com.course.platform.user.dto.UserSummaryDto;

import lombok.*;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class PaymentDetailDto {
    private Integer id;
    private UserSummaryDto student;
    private CourseSummaryDto course;
    private BigDecimal amount;
    private String status;
    private OffsetDateTime paymentDate;
    private String paymentMethod;
}

File: frontend/src/app/domains/payment/models/dto/payment-detail.dto.ts
import { CourseSummaryDto } from '../../../course/models/dto/course-summary.dto';
import { UserSummaryDto } from '../../../user/models/dto/user-summary.dto';

export interface PaymentDetailDto {
  id: number;
  student: UserSummaryDto;
  course: CourseSummaryDto;
  amount: number;
  status: string;
  paymentDate?: string | Date;
  paymentMethod?: string;
}
